srai/regionizers/h3_regionizer.py

```diff
@@ -0,0 +1,191 @@
+            h3_index (str): H3 Index to be converted.
+
+        Returns:
+            geometry.Polygon: Converted polygon.
+
+        """
+        return self._polygon_h3_to_shapely(h3.cells_to_polygons([h3_index])[0])
+
+    def _polygon_h3_to_shapely(self, polygon: h3.Polygon) -> geometry.Polygon:
+        """
+        Convert H3 Polygon to Shapely Polygon.
+
+        Args:
+            polygon (h3.Polygon): H3 Polygon to be converted.
+
+        Returns:
+            geometry.Polygon: Converted polygon.
+
+        """
+        return geometry.Polygon(
+            shell=[coord[::-1] for coord in polygon.outer],
+            holes=[[coord[::-1] for coord in hole] for hole in polygon.holes],
+        )
+
+    def _polygon_shapely_to_h3(self, polygon: geometry.Polygon) -> h3.Polygon:
+        """
+        Convert Shapely Polygon to H3 Polygon.
+
+        Args:
+            polygon (geometry.Polygon): Shapely polygon to be converted.
+
+        Returns:
+            h3.Polygon: Converted polygon.
+
+        """
+        exterior = [coord[::-1] for coord in list(polygon.exterior.coords)]
+        interiors = [
+            [coord[::-1] for coord in list(interior.coords)] for interior in polygon.interiors
+        ]
+        return h3.Polygon(exterior, *interiors)
+
+    def _buffer(self, gdf: gpd.GeoDataFrame, gdf_h3: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+        """
+        Buffer H3 Cells to cover the entire geometries.
+
+        Filling the geometries using h3.polygon_to_cells() with H3 cells does
+        not cover fully the geometries around the borders. To overcome this
+        the approach inspired by [1] is used with the difference, that the neighbouring
+        cells are found for all of the cells, not only the borders.
+
+        Notes:
+            This implementation is quite slow. There is an optimization possible to find
+            only the neighbours for the border cells. It was done that way however,
+            because it is hard to identify contiguous cells and handle this case correctly.
+
+            Other solutions are to buffer the geometries by some constant * edge length,
+            but that gives imprefect solutions due to different H3 cell edge lenghts across
+            the globe.
+
+        Args:
+            gdf (gpd.GeoDataFrame): Geometries.
+            gdf_h3 (gpd.GeoDataFrame): H3 cells.
+
+        Returns:
+            gpd.GeoDataFrame: A superset of original H3 cells with added cells on the borders.
+
+        References:
+            [1] https://stackoverflow.com/a/62519680
+
+        """
+        h3_disk = seq(gdf_h3.index).map(h3.grid_disk).flatten().distinct().to_list()
+        gdf_h3_disk = self._gdf_from_h3_indexes(h3_disk)
+
+        gdf_h3_border = (
+            gdf_h3_disk.sjoin(gdf[["geometry"]], predicate="overlaps")
+            .drop(columns="index_right")
+            .drop_duplicates()
+        )
+
+        return pd.concat([gdf_h3, gdf_h3_border], axis=0).drop_duplicates()
```

Comment:
Will `pd.concat` preserve resulting type as a `GeoDataFrame`?

Reply:
I checked and it is in fact GeoDataFrame, but typing of pd.concat() is ofc wrong